### PR TITLE
Create Plugin: Drop console statements in production builds

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
Introduces a setting to the terser plugin to drop console statements for production builds. Should help reduce dev tools noise and make it easier for developers to achieve 💯  against the validator.

Tested the canary build locally against a dev and prod build:

```typescript
import { AppPlugin } from '@grafana/data';
import { App } from './components/App';
import { AppConfig } from './components/AppConfig';

export const plugin = new AppPlugin<{}>().setRootPage(App).addConfigPage({
  title: 'Configuration',
  icon: 'cog',
  body: AppConfig,
  id: 'configuration',
});

console.log('jack jack app!!!');

```

| Dev | Prod | 
| --- | --- |
| Screenshot 2024-08-20 at 14 24 16 | Screenshot 2024-08-20 at 14 24 32 |

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@5.3.7-canary.1066.fcc7f3a.0
  npm install @grafana/plugin-e2e@1.7.2-canary.1066.fcc7f3a.0
  npm install @grafana/plugin-meta-extractor@0.0.7-canary.1066.fcc7f3a.0
  # or 
  yarn add @grafana/create-plugin@5.3.7-canary.1066.fcc7f3a.0
  yarn add @grafana/plugin-e2e@1.7.2-canary.1066.fcc7f3a.0
  yarn add @grafana/plugin-meta-extractor@0.0.7-canary.1066.fcc7f3a.0
  ```
</details>

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM I guess any real problems should bubble up as exceptions?

**Review by academo — CHANGES_REQUESTED** [review]
If this is removing console log from source maps this will generate problems with the validator and the source code validation

**Review by academo — APPROVED** [review]
tested and can confirm the console logs are only removed in the built files not the source maps.

**Review by sunker — APPROVED** [review]

File: packages/create-plugin/templates/common/_package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "sass-loader": "13.3.1",
     "style-loader": "3.3.3",
     "swc-loader": "^0.2.3",
+    "terser-webpack-plugin": "^5.3.10",
```
Comment (jackw) [inline packages/create-plugin/templates/common/_package.json:54]: We need this updated version of terser to be able to make fine grained choices of which `console.*` methods to drop.

**Pull request merged**